### DISCUSSION TITLE
User documentation?

### DISCUSSION THREAD

**Discussion by sanderant** [jasoncoon/esp8266-fastled-webserver#206] [General]
I have the fibonacci256 and I was wondering if there was any user documentation.

Specifically two things:

Upload firmware:
The link takes me to the release page, but do I have to compile the firmware in Arduino or can I use one of the files from the release page?

Palette:
Is this to be used with the patterns or only some patterns? It doesn't seem to have any affect when I change it.

Thanks for all your hard work!

**Comment by jasoncoon** [comment]
There are pre-compiled binary files for the Fibonacci256 firmware and file system here: https://github.com/jasoncoon/esp8266-fastled-webserver/tree/fibonacci256/bin

Your F256 was fully-assembled, so you should be able to update the firmware and SPIFFS from the Firmware tab in the web UI.

Or you could use something like esptool: https://nodemcu.readthedocs.io/en/latest/flash

Only some patterns will use the selected palette.

Thank you!

**Comment by sanderant** [comment]
Thanks for the clarification. All updated now!

Also seeing your tweet https://twitter.com/jasoncoon_/status/1414245715509977088 clarified things!

Great as always!

**Comment by fretsbiz** [comment]
Hi Jason,

I've been using your esp8266-fastled-webserver for a few years and I really enjoy it. Thank you for all of your hard work! 

Unfortunately, your latest release has me baffled. I can't figure out how to do the SSID & passwd configuration, nor where to enter the number of LED's, their type, and current rating. 

When you have a minute could you please lend me a hand with the new configurations? I run your program on a Lolin Wemos D1 Mini using a strip of (300) WS2812B 's. 

Much obliged, 
Barry Densch

**Comment by jasoncoon** [comment]
Hey Barry, we've recently started to consolidate all of the separate branches and enable support for PlatformIO, which is a major improvement over the Arduino IDE.  It has, however, complicated the code base a bit with a lot of preprocessor directives.

We definitely need to improve the documentation, and maybe rearrange the code to make it easier.  I'll try to work on this soon.

The short answer is that for the default "product", those settings have moved here: https://github.com/jasoncoon/esp8266-fastled-webserver/blob/main/esp8266-fastled-webserver/include/configs/product/default.h

Regarding the wi-fi SSID and password, a while back I moved the project from hard-coding those to using WiFiManager: https://github.com/tzapu/WiFiManager

Now new devices start up their own wi-fi access point.  You connect to it and get directed to a configuration web page where you choose or enter your SSID and password.

**Comment by fretsbiz** [comment]
Much obliged for that info.

I put all of the pieces together last night, and presto! I really like the
AP's clean and simple interface.

Another question/observation. It looks like you are defining the number of
LEDs with a variable that uses the number of pixels instead of the old
num_leds. That was/is a little confusing.

Finally, what is the variable name & location for pin assignment? I ran
across D5 as the default, but changing it to another pin didn't have any
effect.

**Comment by henrygab** [comment]
@fretsbiz -- 

I've written up an answer that details how to configure a different DATA_PIN, plus much more. 

See [this comment](https://github.com/jasoncoon/esp8266-fastled-webserver/discussions/218#discussioncomment-1707936) for details.

**Comment by pixelatedpic** [comment]
how the connections are made? any schematics?

**Reply by henrygab** [reply]
Hi @pixelatedpic ,

You should open a new issue for your question.   Please include additional information, such as:


1. What is your end goal?

1. What hardware?
    * E.g., If the Fibonacci series of hardware, also mention which one (Fib256, Fib128 Nano, etc.)
    * E.g., Else, mention not only the ESP module type, but also which types of pixels you are trying to connect, and what other hardware you already have

1. What have you already tried?

1. What was the **_expected_** result of what you tried?

1. What was the **_actual_** result from your experience?
